### PR TITLE
Pat refactoring

### PR DESCRIPTION
**Pull request by pnaatz**

### CODE REVIEW

**Comment by jcorm714** [comment]
remove unnecessary break statements https://github.com/Pebblz/FindersKeepers/blob/11da841f95e35eb49458e43aae0ff39407b2677f/Assets/Scripts/GameManager.cs#L303

**Comment by jcorm714** [comment]
Merge master into this branch, resolve any conflicts

File: Assets/Scripts/RoomRandomizer.cs [hunk 1]
```diff
@@ -111,11 +112,7 @@ void SpawnRooms()
             //PhotonNetwork.Instantiate(rooms[i].name, roomSpawnpoints[i].transform.position, roomSpawnpoints[i].transform.rotation);
```
Comment (jcorm714) [inline Assets/Scripts/RoomRandomizer.cs:112]: remove commented out code

File: Assets/Scripts/TodoList.cs [hunk 2]
```diff
@@ -67,47 +74,35 @@ public void FillList()
...
+        foreach(Image image in images)
+        {
+            FillImage(image);
+        }
     }
 
+
     void FillImage(Image img)
-    {
+    {//TODO make it so the images don't get repeated
```
Comment (jcorm714) [inline Assets/Scripts/TodoList.cs:90]: remove todo if completed, otherwise fix it

File: Assets/Scripts/TodoList.cs [hunk 3]
```diff
@@ -116,10 +111,15 @@ void FillImage(Image img)
...
+        } else
+        {
+            Debug.Log("tried to load object that doesnt exist");
         }
-        FindObjectOfType<PickUpableSpawner>().FindOBJ();
+
+        FindObjectOfType<PickUpableSpawner>().FindOBJ(); //called to coordinate something with pickupablespawner
     }
 
+    //DIDNT DELETE THIS because it could be used in the future
```
Comment (jcorm714) [inline Assets/Scripts/TodoList.cs:122]: I don't  care remove it, we can always do some git time travel. large blocks of commented out code make reading the file more difficult. Additionally, pickUp has already been changed on master.

File: Assets/Scripts/WinOrLoseScript.cs [hunk 4]
```diff
@@ -54,88 +164,266 @@ void Start()
...
 
+        return incrementation;
+    }
 
-    void Display(Player[] players)
+    IEnumerator Reveal()
     {
-        int incrementation = 0;
+        //properly organize players to drop at same or different times
+        List<List<Player>> order = new List<List<Player>>();
```
Comment (jcorm714) [inline Assets/Scripts/WinOrLoseScript.cs:232]: this seems overly complex for ordering players. Why nested list?

File: Assets/Scripts/WinOrLoseScript.cs [hunk 5]
```diff
@@ -54,88 +164,266 @@ void Start()
...
+            //setup for next loop
+            index--;
+        }
+
+        return podiumsToDestroy;
+    }
+
+    void Order()
+    {
+        for(int current = 0; current < players.Length; current++)
```
Comment (jcorm714) [inline Assets/Scripts/WinOrLoseScript.cs:356]: we already have linq in the project. You could just do  Players.OrderBy(Score)
https://www.tutorialsteacher.com/linq/linq-sorting-operators-orderby-orderbydescending

File: Assets/Scripts/WinOrLoseScript.cs [hunk 6]
```diff
@@ -54,88 +164,266 @@ void Start()
...
+        //otherwise if you didnt win, oyu lost
+        Lose();
     }
-    
-    void Win()
+
+    void Win(Player winner)
     {
-        audioSource.PlayOneShot(winSound);
+        PlaySound(audioVaraibles.winSound);
```
Comment (jcorm714) [inline Assets/Scripts/WinOrLoseScript.cs:387]: we don't have a win/lose sound, these might cause errors temporarily

File: Assets/Scripts/WinOrLoseScript.cs [hunk 7]
```diff
@@ -54,88 +164,266 @@ void Start()
...
+        return false;
+    }
+
+    void PlaySound(AudioClip audioClip)
+    {
+        audioVaraibles.audioSource.PlayOneShot(audioClip);
+    }
+    #endregion
 
     #region Button Functions
```
Comment (jcorm714) [inline Assets/Scripts/WinOrLoseScript.cs:428]: we should really just make a button manager script and put all the button code in there. Also scene loading will not work that way.

**Comment by jcorm714** [comment]
Additionally, the win/lose screen can be simplified, because we decided that there is always 4 podiums. There are some inconsistencies between master and this branch.

File: Assets/Scripts/RoomRandomizer.cs (hunk 1 above)
Comment (pnaatz) [reply Assets/Scripts/RoomRandomizer.cs:112]: uh I did not comment that out

File: Assets/Scripts/TodoList.cs (hunk 3 above)
Comment (pnaatz) [reply Assets/Scripts/TodoList.cs:122]: Brandons highlighting can be put right there if he uncomments it

File: Assets/Scripts/WinOrLoseScript.cs (hunk 6 above)
Comment (pnaatz) [reply Assets/Scripts/WinOrLoseScript.cs:387]: I put the temp sounds in to avoid the error

File: Assets/Scripts/WinOrLoseScript.cs (hunk 7 above)
Comment (pnaatz) [reply Assets/Scripts/WinOrLoseScript.cs:428]: Agreed. I will get to that on a later date.

File: Assets/Scripts/WinOrLoseScript.cs (hunk 5 above)
Comment (pnaatz) [reply Assets/Scripts/WinOrLoseScript.cs]: Completed

File: Assets/Scripts/WinOrLoseScript.cs (hunk 4 above)
Comment (pnaatz) [reply Assets/Scripts/WinOrLoseScript.cs]: Completed

File: Assets/Scripts/TodoList.cs (hunk 2 above)
Comment (pnaatz) [reply Assets/Scripts/TodoList.cs]: Completed

File: Assets/Scripts/Launcher.cs [hunk 8]
```diff
@@ -72,23 +81,32 @@ public void Connect()
...
-    public void Quit()
+    /// <summary>
+    /// Quits the game
+    /// </summary>
+    public void Quit() //add to quit button
     {
         Quit();
     }
     #endregion
+    #endregion
```
Comment (jcorm714) [inline Assets/Scripts/Launcher.cs:109]: two end regions?

**Review by jcorm714 — COMMENTED** [review]
Make sure to merge main onto this branch

**Review by jcorm714 — APPROVED** [review]

File: Assets/Scripts/Launcher.cs (hunk 8 above)
Comment (pnaatz) [reply Assets/Scripts/Launcher.cs:114]: its for a different region

**Pull request merged**